sklearn/manifold/_locally_linear.py

```diff
@@ -401,7 +404,7 @@ def locally_linear_embedding(
             M[nbrs_x, nbrs_y] += np.dot(w, w.T)
 
         if M_sparse:
-            M = csr_matrix(M)
+            M = M.tocsr()
```

Comment:
I assume that we get this outside of all the `if` statement since it would be true for all methods.

Reply:
For the "standard" we could delay the `.tocsr()` then.